test/loki_service.cc

```diff
@@ -423,33 +416,34 @@ void run_requests(const std::vector<http_request_t>& requests,
   client.batch();
 
   // Make sure that all requests are tested
-  test::assert_bool(success_count == requests.size(),
-                    "Expected passed tests count: " + std::to_string(requests.size()) +
-                        " Actual passed tests count: " + std::to_string(success_count));
+  EXPECT_EQ(success_count, requests.size());
 }
 
-void test_failure_requests() {
+TEST(LokiService, test_failure_requests) {
   run_requests(valhalla_requests, valhalla_responses);
 }
 
-void test_osrm_failure_requests() {
+TEST(LokiService, test_osrm_failure_requests) {
   run_requests(osrm_requests, osrm_responses);
 }
-} // namespace
 
-int main(void) {
-  // make this whole thing bail if it doesnt finish fast
-  alarm(180);
+// todo: test_success_requests??
 
-  test::suite suite("Loki Service");
+} // namespace
 
-  // test failures
-  suite.test(TEST_CASE(start_service));
-  suite.test(TEST_CASE(test_failure_requests));
-  suite.test(TEST_CASE(test_osrm_failure_requests));
+class LokiServiceEnv : public ::testing::Environment {
+public:
+  void SetUp() override {
+    start_service();
+  }
+};
 
-  // test successes
-  // suite.test(TEST_CASE(test_success_requests));
+// Elevation service
+int main(int argc, char* argv[]) {
+  // make this whole thing bail if it doesnt finish fast
+  alarm(180);
 
-  return suite.tear_down();
+  testing::AddGlobalTestEnvironment(new LokiServiceEnv);
```

Comment:
not that it matters much but why `new` without a `delete`. you could just instantiate and then get the pointer from the instance so taht at least its distructed when it goes out of scope. again it doesnt matter because its reclaimed when the program ends, but just wondering.

Reply:
The relevant guideline for the entire test environment is taken from here:

https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#global-set-up-and-tear-down

> Note that googletest takes ownership of the registered environment objects. Therefore do not delete them by yourself.

Reply:
btw, in some test files (including this one) some test cases were not included into the suite (and not even commented out). Though their scenarios were pretty valid.